rpcs3/Emu/Cell/lv2/sys_game.cpp

```diff
@@ -5,52 +5,100 @@
 #include "Emu/System.h"
 #include "Emu/system_config.h"
 
-#include <thread>
+#include "Emu/IdManager.h"
+#include "Utilities/Thread.h"
 
 #include "sys_game.h"
 
 LOG_CHANNEL(sys_game);
 
-atomic_t<bool> watchdog_stopped = true;
-atomic_t<u64> watchdog_last_clear;
-u64 get_timestamp()
+struct watchdog_t
 {
-	return (get_system_time() - Emu.GetPauseTime());
-}
+	struct alignas(16) control_t
+	{
+		bool needs_restart = false;
+		bool active = false;
+		char pad[sizeof(u64) - sizeof(bool) * 2]{};
+		u64 timeout = umax;
+	};
 
-error_code _sys_game_watchdog_start(u32 timeout)
-{
-	sys_game.trace("sys_game_watchdog_start(timeout=%d)", timeout);
+	atomic_t<control_t> control{};
 
-	if (!watchdog_stopped)
+	void operator()()
 	{
-		return CELL_EABORT;
-	}
+		u64 start_time = get_system_time();
+		u64 old_time = start_time;
+		u64 current_time = old_time;
 
-	auto watchdog = [=]()
-	{
-		while (!watchdog_stopped)
+		constexpr u64 sleep_time = 50'000;
```

Comment:
I wonder how this is decided.
In the log, I saw the watchdog was cleared each frame (sys_game_watchdog_clear() was called per 16.x ms).

Reply:
But this API has 1s resolution anyway, why it matters?